### ISSUE TITLE
FunctionN::lift and CheckedFunctionN::lift version with Try as return type

### ISSUE THREAD

**Issue by manu-m** [javaslang/javaslang#1288]
Current lift method of FunctionN and CheckedFunctionN returns a partial function that returns an Option. From my point of view it would be nice to have a lift version that returns a Try instead. Why? Because in this case I can get the cause why the function is not defined for specific value(s). I think this makes much sense for further error handling and error processing. 

The API could be defined as follows (to be backward compatible):
```java
<T..., R> Try<R> FunctionN::liftTry(FunctionN<T..., R> partialFunction)
<T..., R> Try<R> CheckedFunctionN::liftTry(FunctionN<T..., R> partialFunction)
```
(maybe you have a better method name in mind :-))

What do you think?

If you have a similar view on this topic, I could try to do a PR (if it isn't too hard, I'm not very familiar with the code generation)

**Comment by danieldietrich** [comment]
Hi Manuel,

I like the idea - this makes sense! The name is great.

Pull requests are welcome :-) If you do not feel comfortable with the generator I can get my hands on it - just give me a sign.

**Comment by manu-m** [comment]
I will make a PR. I am interested in how the generator works and I want to help this great project :-). If the PR doesn't comform to your quality standards you can still get your hands dirty ;-)

**Comment by danieldietrich** [comment]
Sounds great :) Here are some hints:

The generator is a big String-template that expands. The two most essential helpers are:

* _Extended_ multi-line strings: `xs"""..."""`. If've written a function that aligns those Strings automatically. So we need no other pretty printer.
* There are various Strings, Boolean and Range extensions (via implicits). Scroll to the bottom of the generator `javaslang/generator/Generator.scala` - there are they listed. Some parameters are optional, like separators.

```scala
// = <1><2><3>
(1 to 3).gen(i => s"<$i>")

// = <1>
// <2>
// <3>
(1 to 10).gen(i => s"")("\n")

// = test
(1 + 1 == 2).gen("test")

// (does nothing)
(0 == 1).gen("test")

xs"""
  Just text
  ${/* code */}
  text again
"""
```

Scala code is aligned by 2 spaces, generated Java code is aligned by 4 spaces.

**Comment by valery1707** [comment]
This methods will be implemented in #1385

**Comment by danieldietrich** [comment]
Fixed with #1385

/cc @valery1707 thx for the hint!
